lightningd/onchain_control.c

```diff
@@ -518,6 +518,43 @@ static u32 infinite_block_deadline(const struct chain_topology *topo)
 	return get_block_height(topo) + 300;
 }
 
+/**
+ * slow_sweep_dealine -- A smart(-ish) way of estimating sweep fees.
+ *
+ * We use the `close_blockheight` as the basis for the target. We then
+ * target to have the funds back at most 2 weeks after the close,
+ * linearly increasing the confirmation target until we get
+ * close. However, we also don't want to panic and set a target that
+ * is too close, in order not to waste too many funds on the sweep
+ * fees.
+ */
+static u32 slow_sweep_deadline(const struct chain_topology *topo, const struct channel *c)
+{
+	u32 deadline, height = get_block_height(topo);
+	u32 closeheight = *c->close_blockheight ? *c->close_blockheight : 0;
+	if (closeheight == 0)
+		return get_block_height(topo) + 300;
```

Comment:
why not just return `height + 300` here?

note that this is the same as the case on L552 (either your deadline is farther out than 300 blocks or the close tx hasn't been  confirmed yet)

Reply:
Yep, thanks, that's a leftover of refactoring the code several times. I'll reuse the `height`.